core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -321,17 +326,28 @@ protected boolean shouldUpdateStreams(long currentBlockId) {
 
   /**
    * Closes or cancels {@link #mCurrentCacheStream}.
-   *
-   * @throws IOException if the close or cancel fails
    */
-  private void closeOrCancelCacheStream() throws IOException {
+  private void closeOrCancelCacheStream() {
     if (mCurrentCacheStream == null) {
       return;
     }
-    if (mCurrentCacheStream.remaining() == 0) {
-      mCurrentCacheStream.close();
-    } else {
-      mCurrentCacheStream.cancel();
+    try {
+      if (mCurrentCacheStream.remaining() == 0) {
+        mCurrentCacheStream.close();
+      } else {
+        mCurrentCacheStream.cancel();
+      }
+    } catch (IOException e) {
+      if (e.getCause() instanceof BlockDoesNotExistException) {
+        LOG.info("Block {} does not exist when being cancelled.", getCurrentBlockId());
```

Comment:
1. If I use LOG.warn or LOG.error, the spark shell will be spammed by logs unless the user do sc.setLogLevel("ERROR").  In fact, i think this should be LOG.debug. Otherwise it might affect performance since this it happens so often in large parquet files.  
2. Documents added. 
3. This change optimizes for concurrent Alluxio readers with cache turned on (note that it affects concurrent readers with or without partially read block caching turned on). It affects normal cases. We used to throw and exception to the reader if the caching fails (can be many reasons such as network partition, unreachable due to alluxio worker is busy etc. etc. ), but now we don't.   I think it is expected to see alluxio client to behave this way, i.e. don't crash the reader if cache fails.  For example, a spark user might prefer to have his job working instead of crashing if one AlluxioWorker lost.